java/src/main/native/src/RmmJni.cpp

```diff
@@ -365,6 +367,11 @@ JNIEXPORT void JNICALL Java_ai_rapids_cudf_Rmm_initializeInternal(
         Initialized_resource = rmm::mr::make_owning_wrapper<rmm::mr::arena_memory_resource>(
             std::make_shared<rmm::mr::cuda_memory_resource>(), pool_size, pool_limit);
       }
+    } else if (use_cuda_async_alloc) {
+      auto pool_limit = (max_pool_size > 0) ? thrust::optional<std::size_t>{max_pool_size} :
+                                              thrust::optional<std::size_t>{};
+      Initialized_resource =
+          std::make_shared<rmm::mr::cuda_async_memory_resource>(pool_size, pool_limit);
```

Comment:
If we wrap this resource, like so:

```
      Initialized_resource = rmm::mr::make_owning_wrapper<rmm::mr::limiting_resource_adaptor>(
         std::make_shared<rmm::mr::cuda_async_memory_resource>(
           pool_size /* minimum size*/, 
           max_pool_size /* threshold */), 
         max_pool_size);
```

The behavior is the same as the other allocators.

Reply:
It would be great if we can add comments around these arguments. I had some here, feel free to use if you like: https://github.com/rapidsai/cudf/pull/9208/files#diff-92a2a131c5c5590cd59fa8fd86b8bde5aee88ee1c45fb611d661e644e446ea8aR363

Reply:
Added some comments.